x-pack/plugins/apm/common/locator.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { LocatorDefinition } from 'src/plugins/share/common';
+import type { SerializableRecord } from '@kbn/utility-types';
+import * as t from 'io-ts';
+import { isRight } from 'fp-ts/Either';
+
+const serviceOverViewTab = t.keyof({
+  transactions: null,
+  metrics: null,
+  logs: null,
+});
+
+export const APM_APP_LOCATOR = 'APM_LOCATOR';
+
+type ServiceOverViewTab = t.TypeOf<typeof serviceOverViewTab>;
+
+export interface ServiceViewLocatorParams extends SerializableRecord {
+  serviceName?: string;
+  serviceOverViewActiveTab?: ServiceOverViewTab;
+}
+
+export type APMLocatorParams = ServiceViewLocatorParams;
+
+export class APMLocatorDefinition
```

Comment:
Nit: This can just be an object. Also it seems more like an APM service overview locator. If it's meant to only be one locator that will possible handle other pages, the params will have to change and support some kind of page type.

Reply:
I've used a class here mostly for the sake of consistency with the other Kibana plugins, also to be able to use the `LocatorDefinition` interface contract.
About having one locator per feature or a single one that will handle all pages, I wasn't sure about it, since I'm not familiar with all APM-UI pages and features, so I wanted to get some input from the team before deciding on that myself.

Reply:
Instead of declaring routes manually in `APMLocator`, I opted in for generating them based on the presence of  the key 'locatorPageId' in the `APM` route definition, using the [getLocatorEnabledRoutes helper ](https://github.com/elastic/kibana/blob/cbbc8c4202de6d203d544327ef19f4f0ba3799da/x-pack/plugins/apm/public/locator/get_locator_enabled_routes.ts), let me know what do you think about this approach instead!